SimplifiedTests/NYPLAgeCheckTests.swift

```diff
@@ -0,0 +1,129 @@
+//
+//  NYPLAgeCheckTests.swift
+//  Simplified
+//
+//  Created by Ernest Fan on 2021-03-09.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+import XCTest
+@testable import SimplyE
+
+class NYPLAgeCheckTests: XCTestCase {
+
+  // Classes/mocks needed for testing
+  var libraryAccountMock: NYPLLibraryAccountMock!
+  var ageCheckChoiceStorageMock: NYPLAgeCheckChoiceStorageMock!
+  var userAccountProviderMock: NYPLUserAccountProviderMock!
+  var simplyeLibraryAccountProviderMock: NYPLCurrentLibraryAccountProviderMock!
+  var ageCheck: NYPLAgeCheck!
+  
+  // NYPLAgeCheck checks the property userAboveAgeLimit in AccountDetails before performing age check
+  // This property is store in UserDefault which can be different value when testing on different machine
+  // And AccountDetails class is final and not protocol so we cannot override/mock it
+  // The workaround here is to store the value of userAboveAgeLimit and restore it after each test
+  // This way we can set whatever value we need in each tests, and no need to worry about the default value on different machine
+  var defaultUserAboveAgeLimit: Bool!
+  
+  // Use expectation to test result within closure
+  var expectation: XCTestExpectation!
+  
+  override func setUpWithError() throws {
+    try super.setUpWithError()
+    
+    libraryAccountMock = NYPLLibraryAccountMock()
```

Comment:
We can delete libraryAccountMock since it's not used anywhere in the below tests

Reply:
Nice catch!